src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp

```diff
@@ -910,10 +912,18 @@ class G1PostEvacuateCollectionSetCleanupTask2::FreeCollectionSetTask : public G1
     _g1h->collection_set_par_iterate_all(&cl, &_claimer, worker_id);
     // Report per-region type timings.
     cl.report_timing();
-    _g1h->phase_times()->record_or_add_thread_work_item(G1GCPhaseTimes::RestoreRetainedRegions,
-                                                        worker_id,
-                                                        cl.num_retained_regions(),
-                                                        G1GCPhaseTimes::RestoreRetainedRegionsRetainedNum);
+
+    G1GCPhaseTimes* p = _g1h->phase_times();
+    p->record_or_add_thread_work_item(G1GCPhaseTimes::RestoreEvacuationFailedRegions,
+                                      worker_id,
+                                      _evac_failure_regions->num_regions_pinned(),
+                                      G1GCPhaseTimes::RestoreEvacFailureRegionsPinnedNum);
+    p->record_or_add_thread_work_item(G1GCPhaseTimes::RestoreEvacuationFailedRegions,
+                                      worker_id,
+                                      _evac_failure_regions->num_regions_alloc_failed(),
+                                      G1GCPhaseTimes::RestoreEvacFailureRegionsAllocFailedNum);
+
```

Comment:
The counts used here are the totals, but they are added to thread work item, so total count will be wrong.

Discussed this a bit with Thomas and we should probably count those things in this task instead and could that way make `G1EvacFailureRegions` store a bit less information.